javascript/src/seedsource-ui/containers/RunStep.tsx

```diff
@@ -98,12 +100,26 @@ const connector = connect(
     },
 
     onSave: (isLoggedIn: boolean) => {
-      if (!isLoggedIn) {
+      const requestLogin = () => {
         dispatch(setError(t`Login required`, t`Please login to save your run.`))
+      }
+
+      if (!isLoggedIn) {
+        requestLogin()
         return
       }
 
-      dispatch(showSaveModal())
+      get('/accounts/user-info/')
+        .then(response => {
+          if (response.status < 200 || response.status >= 300) {
+            throw new Error()
+          }
+          dispatch(showSaveModal())
+        })
+        .catch(() => {
+          dispatch(logout())
+          requestLogin()
+        })
```

Comment:
This follows a similar pattern [used here](https://github.com/consbio/seedlot-selection-tool/blob/bbf2ea2c16ea33a2dfc95a3b274baf7100544b3d/javascript/src/seedsource-ui/containers/AccountMenu.tsx#L23) to check if a user is logged in.